code/modules/vending/_vending.dm

```diff
@@ -866,8 +866,14 @@ GLOBAL_LIST_EMPTY(vending_products)
 			if(icon_vend) //Show the vending animation if needed
 				flick(icon_vend,src)
 			playsound(src, 'sound/machines/machine_vend.ogg', 50, TRUE, extrarange = -3)
-			new R.product_path(get_turf(src))
+			var/vended = new R.product_path(get_turf(src))
 			R.amount--
+			if(istype(vended, /obj/item))
```

Comment:
Can anything that's not an item be vended?

Reply:
I am not sure, but this prevents any edge cases which would cause runtimes. Better safe than sorry.

Reply:
Silent sanity checks are bad in the long term, as if something does fail we'll have a hard time tracking it.
Just remove the check entirely, we want errors situations to runtime.
And if you must add a sanity check anywhere, do so with an informative `stack_trace` or `CRASH`, as relevant.

Reply:
Agreed, I've updated it.